### ISSUE TITLE
Echo SR video attributes in OpenRTB response

### ISSUE THREAD

**Issue by bszekely1** [prebid/prebid-server#1075]
## Type of issue

Intent to implement

## Description

Prebid Mobile (SDK) is adding video support that includes rendering video using the devices player. Several video specific fields can be set in the Stored Request DB that the device player would not have context to at time of render. The requester (in this case Prebid SDK) can send a signal to Prebid Server, which Stored Request object it expects PBS to respond with in an extension object. 

## Goals

- Supply Prebid Server a signal to respond with specific Stored Request attributes in it's Bid Response

## Proposed Design

Copy all the stored requests attributes for that impression into an extension object called "storedRequestAttributes".

### Prebid Server OpenRTB Changes
The requester will signal to Prebid Server which stored request object it would like to echo back in the Bid Response.

SDK will not expose any features to the publisher facing API. All requesting objects to Prebid Server will be generated from the Prebid SDK itself.

OpenRTB request to Prebid Server
```
{
   ... 
    
    "imp": [
        {
            "banner": {
                "format": [
                    {
                        "h": 250, 
                        "w": 300
                    }
                ]
            }, 
            "ext": {
                "prebid": {
                    "storedrequest": {
                        "id": "xxx"
                    },
                    "options": {
                        "echovideoattrs": true
                    } 
                }
            }, 
            "id": "PrebidMobile", 
            "secure": 1
        }
    ]
    ...
}
```

Proposed OpenRTB response
```
{
    "seatbid": [
        {
            "bid": [
                {
                   ...
                    "ext": {
                        "prebid": {
                            "type": "video",
                            ...
                            },
                            "storedrequestattributes":{
                                "maxduration": 60, 
                                "mimes": [
                                    "video/mp4"
                                ], 
                                "minduration": 15, 
                                "protocols": [
                                    1, 
                                    2
                                ], 
                                "skipafter": 0, 
                                "skipmin": 0, 
                                "startdelay": 0,
                                "playbackmethod": [1]
                            }
                        },
                        "bidder": {
                            "prebid": {
                                "type": "video"
                            }
                        }
                    }
                }
            ]
        }
    ],
..
}
```

### Prebid SDK Changes
Prebid SDK will be responsible for reading any or all attributes. 

### Prebid Server Changes

Prebid Server will be required to parse the Stored Request object to look for video attributes. If one or more attributes are present, echo in OpenRTB extension object in the response. Otherwise, do not include the "storedRequestAttributes" object.

**Comment by bretg** [comment]
Discussed in Prebid Server PMC last week. We suggest ext.prebid.options.echovideoattrs for the flag.

**Comment by mansinahar** [comment]
@bretg We're working on implementing this on the Go side and were wondering if it would make more sense to copy the _resolved_ request's (i.e. the final request after the stored request data has been merged with the incoming request) video object rather than the Stored Request's video object in the response's `prebid.ext.storedRequestAttributes` field? The reason for this is that if there are any video attributes specified in the incoming request then they override the ones in the Stored Request and in that case sending back the values from the Stored Request in `prebid.ext.storedRequestAttributes` would not be the correct thing.

**Comment by bretg** [comment]
@mansinahar -  I'm going to suggest that we stick with what's implemented in PBS-Java and documented at https://docs.prebid.org/prebid-server/endpoints/openrtb2/pbs-endpoint-auction.html#echo-storedrequest-video-attributes-pbs-java-only

1) consistency is good
2) prebid.ext.storedrequestattributes is pretty explicit. It says storedrequest right there

I'm open to re-considering the feature (the original proponent is gone now) and renaming the output to something more like what you're saying... prebid.ext.videoattributes. But since I'm not familiar with how these values are used, I'd need to work with the video taskforce on the definition.

**Comment by SyntaxNode** [comment]
Implemented in PBS-Go 0.171.0.
